src/views/IngredientPage.vue

```diff
@@ -1,9 +1,65 @@
 <template>
-  <main>
-    <TheWelcome content="Ingredient Page" />
-  </main>
+  <div class="h-screen w-screen bg-gray_01-light dark:bg-gray_01-dark">
+    <section
+      class="mx-auto flex h-full flex-col items-center justify-between bg-gray_00-light text-gray_05-light dark:bg-gray_00-dark dark:text-gray_05-dark sm:w-screen md:w-96"
+    >
+      <ProgressNavBar prevPage="/size" pageName="ingredient" />
+      <div v-if="isLoading">로딩중입니다.</div>
+      <IngredientBoard :ingredients="data?.ingredients ?? []" v-else />
+      <div v-if="!isAbleToRecommend">재료를 조금 더 골라볼까요?</div>
+      <NextButton
+        :onClick="
+          () => {
+            mutation.mutate(userItem);
+          }
+        "
+        isPrimary
+        :disabled="!isAbleToRecommend || !userSelect.sizeValue"
+      >
+        결과
+      </NextButton>
+    </section>
+  </div>
 </template>
 
 <script setup lang="ts">
-import TheWelcome from '@components/TheWelcome.vue';
+import { computed } from 'vue';
+
+import ProgressNavBar from '@containers/ProgressNavBar.vue';
+import IngredientBoard from '@containers/IngredientBoard.vue';
+import NextButton from '@components/NextButton.vue';
+import { useUserSelectStore } from '@store/storeUserSelect';
+import { useGetIngredients } from '@apis/ingredients';
+import { usePostRecipe } from '@apis/recipes';
+
+const store = useUserSelectStore();
+const { userSelect } = store;
+const userItem = {
+  sizeId: userSelect.sizeId,
+  ingredientIds: userSelect.ingredientIds,
+};
+
+const { data, isLoading } = useGetIngredients();
+const mutation = usePostRecipe();
+
+const allFlavorIdList = computed(
+  () =>
+    userSelect.ingredientIds
+      .map((id) => {
+        const flavorIdList = data.value?.ingredients.find(
+          (ingredient) => ingredient.id === id
+        )?.flavorIdList;
+        return flavorIdList;
+      })
+      .flat() ?? []
+);
+const flavorIdList = computed(
+  () =>
+    allFlavorIdList.value.filter(
+      (item, index) => allFlavorIdList.value.indexOf(item) === index
+    ) ?? []
+);
+const isAbleToRecommend = computed(
+  () => flavorIdList.value.length >= userSelect.sizeValue
+);
```

Comment:
computed는 보통 state의 get, set의 경우에만 사용합니다.
map, filter 처럼 복잡한 계산을 필요로 하는 함수는 methods로 사용해주시면 되요.
vue3에서 methods란 setup 내부에서 일반 화살표 함수처럼 사용해주시면 됩니다.
```suggestion
const allFlavorIdList =  () => {
    userSelect.ingredientIds
      .map((id) => {
        const flavorIdList = data.value?.ingredients.find(
          (ingredient) => ingredient.id === id
        )?.flavorIdList;
        return flavorIdList;
      })
      .flat() ?? []
      };
const flavorIdList = 
  () => {
    allFlavorIdList.value.filter(
      (item, index) => allFlavorIdList.value.indexOf(item) === index
    ) ?? []
};
const isAbleToRecommend = computed(
  () => flavorIdList.value.length >= userSelect.sizeValue
);
```

Reply:
userSelect에 대해 reactivity를 유지하기 위해서 computed를 사용했는데, 위와 같이 함수로 작성해도 동작을 하네요!!
다만 혹시 method로 작성하게 되면 사용할 때마다 호출을 해야한다는 단점이 있는 것도 같아요..!
관련해서 좀더 찾아보도록 하겠습니다 :)

Reply:
@Air-ljh8 애초에 반응성으로 생성한 state들은 (ref, reactive) computed를 해주지 않아도 되는 걸로 알고 있습니다. 제가 잘못 인지하고 있는 것일 수도 있으니 다른 분들의 의견도 고려해주세요!